apps/api/src/types/index.ts

```diff
@@ -544,6 +544,40 @@ export const CompetitionStatusSchema = z.enum(COMPETITION_STATUS_VALUES);
  */
 export type CompetitionStatus = z.infer<typeof CompetitionStatusSchema>;
 
+/**
+ * Competition agent status values for zod or database enum.
+ */
+export const COMPETITION_AGENT_STATUS_VALUES = [
```

Comment:
@mzkrasner i'm gonna give this another pass and test it out locally, so i'll probably have some additional feedback. but overall, lookin good so far.

one thing that i think we should change is the per-competition statuses:
- `active` => the agent is (or _was_) active in the competition
- `left` => rename to `withdrawn` for semantic clarity
- `removed` => rename to `disqualified` for semantic clarity

the current logic sets an agent to `inactive` on a per-competition basis upon conclusion. i think that can cause confusion and is unnecessary. if a competition is "ended", then we know the per-agent's competition status would reflect the "parent's" state.